tests/test_builder.py

```diff
@@ -416,6 +416,18 @@ def test_data_build_invalid_x_escape(self) -> None:
         with self.assertRaises(AstroidSyntaxError):
             self.builder.string_build('"\\x1"')
 
+    def test_data_build_error_filename(self) -> None:
+        """Check that error filename is set to modname if given."""
+        with pytest.raises(AstroidSyntaxError, match="invalid escape sequence") as ctx:
+            self.builder.string_build("'\\d+\\.\\d+'")
+        assert isinstance(ctx.value.error, SyntaxError)
+        assert ctx.value.error.filename == "<unknown>"
```

Comment:
Not exactly sure why the `SyntaxError` is raised for all Python version. Additionally this should only emit a `SyntaxWarning` at the moment, not a `SyntaxError` (which is reraised as `AstroidSyntaxError`).

Might be some pytest magic though..